content/en/docs/what-is-grpc/core-concepts.md

```diff
@@ -197,6 +197,12 @@ terminates the RPC immediately so that no further work is done.
 Metadata is information about a particular RPC call (such as [authentication
 details](/docs/guides/auth/)) in the form of a list of key-value pairs, where the
 keys are strings and the values are typically strings, but can be binary data.
+
+Keys must consist of lowercase letters, digits, and special characters `-`, `_`,
```

Comment:
Many of the gRPC APIs require lowercase letters, but it is really just an implementation convenience/optimization. For a concept document, I'd say the keys are case insensitive. If languages require lower-case that is an API decision, and it doesn't disagree with the concept. So I'd probably say:

> Keys are case insensitive and consists of ASCII letters, digits, ...

I'd probably rephrase the second sentence to be less specification-y and more of an explanation:

> Binary-valued keys end in `-bin` while ASCII-valued keys do not.

(Or some such.)

Reply:
>  but it is really just an implementation convenience/optimization.

Oh, I misunderstood about it. I fixed the phrases as you proposed 👍